From: SeongJae Park <sj@kernel.org>
Date: Thu, 4 Sep 2025 20:38:43 -0700
Subject: [PATCH] ==== hotfix of refresh_ms ====

mm/damon/sysfs: fix refresh_ms control overwriting on multi-kdamonds usages

Automatic esssential DAMON/DAMOS status update feature of DAMON sysfs
interface (refresh_ms) is broken [1] for multiple DAMON contexts
(kdamonds) use case, since it uses a global single damon_call_control
object for all created DAMON contexts.  The fields of the object,
particularly the list field is over-written for the contexts and it
makes unexpected results including user-space hangup and kernel crashes
[2].  Fix it by extending damon_call_control for the use case and
updating the usage on DAMON sysfs interface to use per-context
dynamically allocated damon_call_control object.

[1] https://lore.kernel.org/20250904011738.930-1-yunjeong.mun@sk.com
[2] https://lore.kernel.org/20250905035411.39501-1-sj@kernel.org

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus__sbag1p8 | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus__sbag1p8

diff --git a/hkml_cv_bogus/hkml_cv_bogus__sbag1p8 b/hkml_cv_bogus/hkml_cv_bogus__sbag1p8
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

